### PR TITLE
Add support for callback to arbitrarily format the server response

### PR DESCRIPTION
**Pull request by nick-giudici**
We have a use case where an existing server is returning a single object with a key value pair for each autocomplete item. Since we don't have control over the response format of the endpoint, we need something that will allow us reformat the response.

This adds an optional callback that allows for arbitrary reformatting of the data returned from the endpoint.

### CODE REVIEW

File: src/components/Autocomplete.vue [hunk 1]
```diff
@@ -139,6 +139,13 @@ export default {
       default: 'name'
     },
 
+    /**
+     * Callback to format the server data
+     */
+    setResults: {
```
Comment (charliekassel) [inline src/components/Autocomplete.vue]: perhaps this prop could be called something like `customSetResults`?

File: src/components/Autocomplete.vue [hunk 2]
```diff
@@ -281,7 +288,11 @@ export default {
           throw new Error('Network response was not ok.')
         })
         .then(response => {
-          this.results = this.setResults(response)
+          if(this.setResults){
```
Comment (charliekassel) [inline src/components/Autocomplete.vue]: and then this logic could move inside the originally named `setResults` function

**Review by charliekassel — COMMENTED** [review]
Thank you for this PR - this is a useful addition.
I think we should just make a single function call inside the resolved promise to set the results so it's easier to read.
The event emitting could also be extracted to a method too, but that's not touching your addition so I'm happy to do that.

File: src/components/Autocomplete.vue (hunk 2 above)
Comment (nick-giudici) [reply src/components/Autocomplete.vue]: That's a reasonable suggestion. I've added it to my fork.

**Review by charliekassel — APPROVED** [review]

**Comment by charliekassel** [comment]
I've renamed the prop to `resultsFormatter` and added a test to document usage: d45441a
Available from v0.9.0

**Pull request merged**